src/Compilation.zig

```diff
@@ -2302,6 +2305,7 @@ fn addNonIncrementalStuffToCacheManifest(comp: *Compilation, man: *Cache.Manifes
 
     man.hash.addListOfBytes(comp.clang_argv);
 
+    man.hash.addOptionalBytes(comp.bin_file.options.entitlements);
```

Comment:
there's a "Mach-O specific stuff" section about 30 lines below. Also this is a file so it needs to be added to the manifest as a file dependency:

```suggestion
    man.hash.addOptionalFile(comp.bin_file.options.entitlements);
```

Same deal with the other cache API usage in the linker.